src/webgpu/api/operation/vertex_state.spec.ts

```diff
@@ -171,8 +184,19 @@ class IndexFormatTest extends GPUTest {
       1.0, 1.0, 0.0, 1.0,
       -1.0, -1.0, 0.0, 1.0]
     );
+
+    let bufferSize = vertexArray.byteLength;
+    // Create a large enough buffer for uint16 primitive restart value used as a big index.
+    if (
+      indexFormat === 'uint16' &&
+      primitiveTopology !== 'line-strip' &&
+      primitiveTopology !== 'triangle-strip'
+    ) {
```

Comment:
You can just unconditionally allocate the buffer to have exactly `2**16` vertex elements regardless of the parameters. Since that's large-ish (1MiB) it would be good to explicitly destroy the buffer at the end of this function after the submit.